scripts/envgen.js

```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fs = require('fs/promises');
+
+const environment = (process.env.NODE_ENV ?? 'development').toLowerCase();
+const path = `${__dirname}/..`;
+
+function stringifyOnSingleLine(json) {
+    const object = JSON.parse(json);
+    return JSON.stringify(object);
+}
+
+function asASingleVariable() {
```

Comment:
async function? callbacks are icky

Reply:
We are using CommonJS modules, so we don't have access to top-level await unfortunately. I can turn this into an async function but it will not be called asynchronously...

Reply:
True, but you dont need to right? This is already not awaiting the result of the promises. If you mark the method async and then just call if in the root that will be fine i believe. The process wont exit till the promises are resolved either way